src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs

```diff
@@ -543,9 +551,13 @@ private async Task<IActionResult> EditPOST(string contentItemId, string returnUr
             {
                 return RedirectToAction("Edit", new RouteValueDictionary { { "ContentItemId", contentItem.ContentItemId }, { "returnUrl", returnUrl } });
             }
+            else if (createNew)
+            {
+                return RedirectToAction("Create", new RouteValueDictionary { { "id", contentItem.ContentType }, { "returnUrl", returnUrl } });
+            }
             else
             {
-                return LocalRedirect(returnUrl);
+                return Redirect(returnUrl);
```

Comment:
Now that we use @Context.Request.Headers["Referer"].ToString() as the returnUrl. This url could return the absolute url containing the domain and else. Also could happen to return a different domain name than the actual one. I need to add some logic to make sure that it never happens. At least I could truncate the domain part of the Url.